### ISSUE TITLE
Clarify reactive power

### ISSUE THREAD

**Issue by narve** [euniversal/umei-api-specification#69]
Pr comments from Claudia / Vladimir: 

The current implementation of the "MeterReading" resource has only one "power" field. In the modifications pushed by Vladimir, there are 3 different items: "activePower", "reactiveInductivePower" and "reactiveCapacitativePower". Will those 3 informations be submitted simultaneously ? In that case we may need of change the implementation of the "MeterReading" resource. 

We should clarify and remedy these shortcomings.

**Comment by ArnaudDebray** [comment]
The best option for me to keep the spec as generic as possible is indeed to have 3 fields in the meterReadings: "activePower", "reactiveInductivePower" and "reactiveCapacitativePower". 

A similar split of the power in the definition of the bids is however not needed as the quantity type is already specified in the market definition.

**Comment by narve** [comment]
We're still awaiting some clarifications from technical people (electrical engineers etc) on our side. Will update this issue when we have more information.

**Comment by narve** [comment]
We have discussed this internally now. There are two seperate issues/problems/discussion points.

**Comment by narve** [comment]
First point: 

There are several different "products" or QuantityType's: Power, ReactivePower, Energy, Capacity, possibly more at a later stage. 
Each of these quantities are offered (Order), traded (Trade), predicted (BaseLineInterval) and measured (MeterReading). 

Orders and Trades have a reference to the market, which has a QuantityType, which tells us what the "quantity" refers to. 

BaselineInterval and MeterReading do not have QuantityType, they have only a single "Power" property. Herein lies the problem. 

We could add ReactivePower to BaselineInterval and to MeterReading. But we have the same 
problem with Energy and Capacity - should we add those fields as well? 

A more generic approach, which we think would be better, would be to rename MeterReading.Power and BaselineInterval.Power to "Quantity", and add a new property "QuantityType". 
QuantityType would then indicate whether the quantity refers to Power (MW), Energy (MWh), ReactivePower (MVAr) or something else. 


If a portfolio trades in both Power and ReactivePower, it would then need to have 
two baselines, one for Power and one for ReactivePower. Similarly with MeterReadings. 


This also aligns with what our DSO expert tells us: Sometimes the "power" trading happens 
on a different place in the grid than the "reactive power trading. Thus it is beneficial 
to have the possibility of splitting those two quantities in different objects. 

Note that for CSV imports, multiple quantities can be specified as separate columns in a single line. The import mechanism would then convert each line into one or more MeterReadings/BaselineIntervals, as necessary.

**Comment by narve** [comment]
Second point: 

For active power, we currently have a single property "power", which is either negative (consumption) or positive (production). Basically, ReactiveInductivePower is the same as 
consumption of reactive power, while ReactiveCapacitivePower is the same as production of 
reactive power. While we could split "power" (or the new name Quantity) into production and consumption (which is actually what NODES does internally), we should at least be consistent. 
So we prefer to be consistent and have a single field, Quantity, that is either negative or positive. The sign of the number, combined with the QuantityType, tells us exactly what the 
number represents.

**Comment by narve** [comment]
What do you think, @ArnaudDebray ?

**Comment by narve** [comment]
After internal discussion in NODES, we've decided that we would like to go for what I hinted at in previous comments: 

- One field called Quantity, and one field called QuantityType. (Unit is defaulted based on QuantityType). Negative values are used for consumption. 
- If a portfolio is active in multiple markets with different quantity types, it will need to provide separate baselines for each quantity type
- Similarly for meter readings - one meter reading for each quantity type. 


Some of the arguments in favour of using separate rows for each quantity type: 

- Different quantity types are traded in different markets and may have different requirements with regard to interval length (e.g  power trading might happen on an hourly basis, while reactive power might be traded in 15minutes intervals)
- Different markets might have different rules wrt when the baselines must be created (e.g. one day before for energy, one hour before for reactive power), and for when they can be updated, how late they can be updated, whether they can be updated after a trade has been agreed upon, etc
- Baselines and meter readings might potentially be provided by different providers (e.g. a utility provides reactive power readings while the FSP itself provides energy readings)
- This solution will be more future proof if additional quantity types are added
- This solution will be more consistent, so meter readings, baselines, trades, settlement and orders (price points) all look similar and are simpler to compare
-
